### DISCUSSION TITLE
fastify + nginx unit

### DISCUSSION THREAD

**Discussion by kurtextrem** [fastify/fastify#5062] [General]
[Nginx Unit](https://unit.nginx.org/keyfeatures/#performance) (not to be confused with Nginx), has remarkable claims on their website - and at least with PHP apps, their claims seem to be true: https://news.ycombinator.com/item?id=37453854 (see top comment, 3 links). Some commentor mentions 40% higher perf.

So, while googling quite a bit, I found nothing - literally nothing about Nginx Unit & Nodejs. This either means: No one knows about it, no one found it useful or it doesn't work. Judging from the [NPM downloads](https://www.npmjs.com/package/unit-http), it seems to be the former (at 109 weekly downloads).

So, has anyone ever tried out Nginx Unit with Fastify before, has anyone done benchmarks or just any experiences? I figured I'd ask here, because Fastify draws a lot of performance minded folks here, so if not here, I don't know where else we could look for nginx unit nodejs users.

**Comment by metcoder95** [comment]
This is the very first time I heard about it, I imagine that by how's designed using WASM it might be performant compared to plain execution (they might be doing what [Web Containers](https://blog.stackblitz.com/posts/introducing-webcontainers/) does).

It seems (as always) that they claim to support JS server apps as they have an example with [Express](https://blog.stackblitz.com/posts/introducing-webcontainers/).

Maybe someone from the community or the team heard already about it and tried out?

**Comment by jsumners** [comment]
It's basically a "servlet container". It replaces core modules with its own as some sort of "next gen" cgi-bin/wsgi/fastcgi. Personally I haven't seen a use case for it. Modern runtimes provide their own system process that exposes network sockets without having to have some sort of host translating network requests to app requests. Just put a reverse proxy in front of things.
